internal/controller/nginx/config/base_http_config.go

```diff
@@ -12,6 +12,9 @@ var baseHTTPTemplate = gotemplate.Must(gotemplate.New("baseHttp").Parse(baseHTTP
 
 type httpConfig struct {
 	DNSResolver             *dataplane.DNSResolverConfig
+	AccessLog               *dataplane.AccessLog
+	DefaultAccessLogPath    string
+	DefaultLogFormatName    string
```

Comment:
I don't think these fields belong in this type.

I'd recommend moving these to the AccessLog:
```
type AccessLog struct {
    Format     string  // User's format string
    Disabled   bool    // User's disabled flag
    Path       string  // Where to write logs (/dev/stdout)
    FormatName string  // Internal format name (ngf_user_defined_log_format)
}
```
And populate the constants in buildAccessLog().

Reply:
We still need to remove these fields from this type